plugins/woocommerce/tests/e2e-pw/tests/merchant/product-reviews.spec.js

```diff
@@ -196,6 +199,112 @@
 			).toBeVisible();
 		} );
 
+		test( 'can approve a product review', async ( { page, reviews } ) => {
+			const review = reviews[ 0 ]; // Select the first review for approval
+
+			// Go to the Product Reviews page
+			await page.goto(
+				`wp-admin/edit.php?post_type=product&page=product-reviews`
+			);
+
+			// Locate the review to be approved
+			const reviewRow = page.locator( `#comment-${ review.id }` );
+
+			// Ensure the review is not already approved
+			const approveButton = reviewRow.getByRole( 'button', {
+				name: 'Approve',
+			} );
+
+			// Hover to reveal action buttons
+			await reviewRow.hover();
+
+			// Click the "Approve" button and wait for the state to become 'networkidle'
+			await approveButton.click();
+
+			// Verify the review is approved by checking the presence of "Unapprove" button
+			const unapproveButton = reviewRow.getByRole( 'button', {
+				name: 'Unapprove',
+			} );
+			await expect( unapproveButton ).toBeVisible();
+		} );
+
+		test( 'can mark a product review as spam', async ( {
+			page,
+			reviews,
+		} ) => {
+			const review = reviews[ 0 ]; // Select the first review to mark as spam
+
+			// Go to the Product Reviews page
+			await page.goto(
+				`wp-admin/edit.php?post_type=product&page=product-reviews`
+			);
+
+			// Locate the review to be marked as spam
+			const reviewRow = page.locator( `#comment-${ review.id }` );
+			await reviewRow.hover();
+
+			// Select Spam action and verify the review is marked as spam
+			await reviewRow.getByRole( 'button', { name: 'Spam' } ).click();
+
+			// Navigate to the "Spam" tab to verify the review is marked as spam
+			await page.click( 'a[href*="comment_status=spam"]' );
+
```

Comment:
To avoid a false negative, I would add a check here to first make sure that we're on the Spam tab. If the tab switch did not happen yet and the review was not marked as spam the test will pass.

Reply:
Resolved with [729c2cd](https://github.com/woocommerce/woocommerce/pull/50060/commits/729c2cda7e3d97170afb2476cce11c265399d846)